src/extension-consumer/src/produce/mod.rs

```diff
@@ -1,149 +1,269 @@
 use std::fs::File;
 use std::io::{BufReader, BufRead};
-use std::path::PathBuf;
+use std::path::{PathBuf, Path};
 use structopt::StructOpt;
+use tracing::debug;
 
 use fluvio::{Fluvio, TopicProducer};
 use fluvio_types::print_cli_ok;
 use crate::common::FluvioExtensionMetadata;
-use crate::Result;
-
-/// Produce log configuration parameters
-#[derive(Debug)]
-pub struct ProduceLogConfig {
-    pub topic: String,
-    pub partition: i32,
-    pub continuous: bool,
-}
-
-#[derive(Debug)]
-pub enum FileRecord {
-    Lines(PathBuf),
-    Files(Vec<PathBuf>),
-}
+use crate::{Result, ConsumerError};
 
 // -----------------------------------
 // CLI Options
 // -----------------------------------
 
+/// Write messages to a topic/partition
+///
+/// When no '--files' are provided, the producer will read from 'stdin'
+/// and send each line of input as one record.
+///
+/// If one or more files are given with '--files', each file is sent as
+/// one entire record.
+///
+/// If '--key-separator' or '--json-path' are used, records are sent as
+/// key/value pairs, and the keys are used to determine which partition
+/// the records are sent to.
 #[derive(Debug, StructOpt)]
 pub struct ProduceLogOpt {
     /// The name of the Topic to produce to
     #[structopt(value_name = "topic")]
     pub topic: String,
 
-    /// The ID of the Partition to produce to
```

Comment:
why these options are removed?

Reply:
Partition is now auto-assigned, and "continuous" was not being used.

Reply:
Yes, make sense for `partition` but why remove `disable_continuous`?

Reply:
`disable_continuous` is a consumer flag, `-d`, I did not remove that. What I removed here was the `--continuous` flag for producer, which had no code usages, it was dead code.

Reply:
k